File: SGU/191.java
import java.util.*;
import java.io.*;
import java.math.*;

public class Solution {
	public static void main (String[] argv) throws IOException {
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		char c;
		char[] a = new char[30001];
		int idx = 0;
		String str;
		str = in.readLine();
		c = str.charAt(0);
		str = in.readLine();
		int len = str.length();
		for (int i = 0; i < len; ++i) {
			if (idx == 0) {
				idx = 1;
				a[1] = str.charAt(i);
			} else {
				if (str.charAt(i) == a[idx]) --idx;
				else a[++idx] = str.charAt(i);
			}
		}
		if (idx > 1 || a[1] == c) System.out.println("NO");
		else System.out.println("YES");
	}
}

File: SGU/191.cpp
//ac

#include <iostream>
#include <cstdio>
#include <cstring>
#include <string>
using namespace std;

int main() {
	char a[30001];
	int idx;
	char c;
	string str;
	cin >> c >> str;
	idx = 0;
	int len = str.length();
	for (int i = 0; i < len; ++i) {
		if (idx == 0) {
			idx = 1;
			a[1] = str[i];
		} else {
			if (str[i] == a[idx]) --idx;
			else a[++idx] = str[i];
		}
	}
	if (idx > 1 || a[1] == c) cout << "NO" << endl;
	else cout << "YES" << endl;
	return 0;
}

